QIR-README.md

```diff
@@ -0,0 +1,188 @@
+
+A sample Q# file suitable that contains both segments suitable for use with the QIR tool is included as
+`examples/QIR/QirTarget.qs`.
+As described above, this file should be included in all QIR generation builds.
+
+The sample file does not define intrinsics for the full set of Q# primitives.
+You may need to extend this file if it is missing intrinsics you require.
+
+### Target-level instruction set
+
+The target-level instructions can all be defined in a single namespace, or in many namespaces.
+Both samples define them in the `Microsoft.Quantum.Instructions` namespace, but that can be
+changed if desired.
+The actual namespace name is not significant.
+
+Target-level instruction definitions follow this pattern:
+
+```qsharp
+    @Intrinsic("x")
+    operation PhysX (qb : Qubit) : Unit
+    {
+        body intrinsic;
+    }
+```
+
+- The `Intrinsic` attribute is required. The string argument, with a `quantum.qis.` prefix,
+  becomes the name of the global LLVM function that implements this instruction.
+- The signature of the LLVM function is derived directly from the Q# operation's signature.
+- Target-level operations should only have `body` specializations, not `adjoint` or `controlled`.
+- The name of the target-level instructions can be arbitrary. By keeping it different from the
+  name of the user-level instruction, we remove the need to explicitly qualify the name with the
+  namespace when we refer to it in the user-level instruction definition.
+- At the moment, the intrinsic can only have a `body` specialization. This may change in the
+  future, but also might not.
+
+### User-level instruction set
+
+Generally, user-level instructions should be defined in the `Microsoft.Quantum.Intrinsic`
+namespace to match standard Q# usage.
+Similarly, the instruction names should match the names in the standard Q# library in that
+namespace, so that other existing Q# code will find the instructions correctly.
+
+User-level instruction definitions follow this pattern:
+
+```qsharp
+    @Inline()
+    operation X(qb : Qubit) : Unit
+    is Adj {
+        body
+        {
+            PhysX(qb);  
+        }
+        adjoint self;
+    }
+```
+
+- The `Inline` attribute is optional. If it appears, the QIR generator will in-line the
+  implementation of the user-level instruction, so that the generated QIR directly calls the
+  target-level instruction. Otherwise the generated QIR will call into the LLVM function that
+  implements the user-level instruction, which will contain a call into the target-level
+  instruction.
+- The implementation of the user-level instruction can contain arbitrary Q# code. For instance,
+  both sample files define the adjoint of a rotation as the rotation by the negative of the angle.
+  Once I add support for controlled, the controlled `X` implementation will have Q# code that will
+  start by checking the number of control qubits and either calling a target-level `CNOT` if 
```

Comment:
> Once **I add support** for controlled

Consider "Once **the support is added** for controlled"

Reply:
I completely forgot about the readme, thanks! I'll go and revise it more broadly since the instructions for using it when it wasn't integrated yet are obsolete.

Reply:
done.